modules/adtrgtmeBidAdapter.js

```diff
@@ -1,332 +1,358 @@
-      appendImpObject(bid, payloadClone);
+    return bR.map((b) => {
+      const data = createORTB(aR, b);
+      appendImp(b, data);
 
-      return generateServerRequest({payload: payloadClone, requestOptions, bidderRequest: bid});
+      return createRequest({
+        data,
+        options,
+        bidderRequest: b,
+      });
     });
   },
 
-  interpretResponse: function(serverResponse, { data, bidderRequest }) {
-    const response = [];
-    if (!serverResponse.body || !Array.isArray(serverResponse.body.seatbid)) {
-      return response;
+  interpretResponse: function (sR, { data, bidderRequest }) {
+    const res = [];
+    if (!sR.body || !Array.isArray(sR.body.seatbid)) {
+      return res;
     }
 
-    let seatbids = serverResponse.body.seatbid;
-    seatbids.forEach(seatbid => {
-      let bid;
-
+    sR.body.seatbid.forEach((sb) => {
       try {
-        bid = seatbid.bid[0];
+        let b = sb.bid[0];
+
+        res.push({
+          adId: deepAccess(b, 'adId') ? b.adId : b.impid || b.crid,
+          ad: b.adm,
+          adUnitCode: bidderRequest.adUnitCode,
+          requestId: b.impid,
+          cpm: b.price,
+          width: b.w,
+          height: b.h,
+          mediaType: BANNER,
+          creativeId: b.crid || 0,
+          currency: b.cur || DEFAULT_CUR,
+          dealId: b.dealid ? b.dealid : null,
+          netRevenue: true,
+          ttl: getTtl(bidderRequest),
+          meta: {
+            advertiserDomains: b.adomain || [],
+            mediaType: BANNER,
+          },
+        });
       } catch (e) {
-        return response;
+        return res;
       }
-
-      let cpm = bid.price;
-
-      let bidResponse = {
-        adId: deepAccess(bid, 'adId') ? bid.adId : bid.impid || bid.crid,
-        ad: bid.adm,
-        adUnitCode: bidderRequest.adUnitCode,
-        requestId: bid.impid,
-        cpm: cpm,
-        width: bid.w,
-        height: bid.h,
-        creativeId: bid.crid || 0,
-        currency: bid.cur || DEFAULT_CURRENCY,
-        dealId: bid.dealid ? bid.dealid : null,
-        netRevenue: true,
-        ttl: getTtl(bidderRequest),
-        mediaType: BANNER,
-        meta: {
-          advertiserDomains: bid.adomain,
-          mediaType: BANNER,
-        }
-      };
-
-      response.push(bidResponse);
     });
 
-    return response;
+    return res;
   },
-
-  getUserSyncs: function(syncOptions, serverResponses, gdprConsent = {}, uspConsent = '') {
-    const bidResponse = !isEmpty(serverResponses) && serverResponses[0].body;
-    if (bidResponse && bidResponse.ext && bidResponse.ext.pixels) {
-      return extractUserSyncUrls(syncOptions, bidResponse.ext.pixels);
+  getUserSyncs: function (options, sR) {
+    const s = [];
+    if (!options.pixelEnabled && !options.iframeEnabled) {
+      return s;
+    }
+    if (Array.isArray(sR)) {
+      sR.forEach((response) => {
+        const p = response.body?.ext?.pixels;
```

Comment:
Getting the user syncs off the response is a bit risky, consent may have changed